### PR TITLE
Improve testing for convolutions (NFC)

### PR DESCRIPTION
**Pull request by chelini**
- Make some CHECKs more constrained

### CODE REVIEW

File: test/TPP/transform/transform-convolutions.mlir [hunk 1]
```diff
@@ -269,7 +274,7 @@ transform.sequence failures(propagate) {
     %4 = transform.structured.map_linalg_to_tpp filter{["tpp.relu"]} in %3
 
     // Fuse relu and conv on the three outermost loops
-    %5, %loop:3 = transform.structured.fuse %4 { tile_sizes = [1, 1, 1, 0, 0] }
+    %5, %loop:5 = transform.structured.fuse %4 { tile_sizes = [1, 1, 1, 1, 1] }
```
Comment (rengolin) [inline test/TPP/transform/transform-convolutions.mlir:277]: What's the point of this change?

File: test/TPP/transform/transform-convolutions.mlir [hunk 2]
```diff
@@ -285,3 +290,60 @@ transform.sequence failures(propagate) {
...
+    ^bb0(%in: f32, %out: f32):
+      %5 = arith.addf %in, %out : f32
+      linalg.yield %5 : f32
+  } -> tensor<1x56x56x256xf32>
+  return %4 : tensor<1x56x56x256xf32>
+}
+
+// X = linalg.conv_2d_nhwc_hwcf
+// Y = linalg.conv_2d_nhwc_hwcf
+// add(X, Y)
```
Comment (rengolin) [inline test/TPP/transform/transform-convolutions.mlir:324]: If this is a mock resnet example, the pattern is, in its most simplified form:
```
conv -> relu -> conv -> relu -> add -> conv -> relu
          \---------------------^
```

Comment (chelini) [reply test/TPP/transform/transform-convolutions.mlir:324]: yes, it is a simplified version.

File: test/TPP/transform/transform-convolutions.mlir (hunk 1 above)
Comment (chelini) [reply test/TPP/transform/transform-convolutions.mlir:277]: I want to fuse the elementwise op entirely.

**Comment by chelini** [comment]
This commit requires: https://github.com/plaidml/tpp-mlir/pull/215

File: test/TPP/transform/transform-convolutions.mlir [hunk 3]
```diff
@@ -285,3 +290,60 @@ transform.sequence failures(propagate) {
...
+   
+    // We need this to canonicalize pack(unpack)...
+    %func = transform.structured.match ops{["func.func"]} in %arg1 
+    // TODO: bubbling + canonicalization should be a proper transform op, at which point we will be
+    // able to preserve the handles and avoid rematching below.
+    transform.structured.packing_propagation %func
+
+    %5 = transform.structured.match ops{["linalg.generic"]} in %arg1
+    %add_packed = transform.structured.map_linalg_to_tpp filter{["tpp.add"]} in %5
+    %6, %loop:5 = transform.structured.fuse %add_packed { tile_sizes = [1, 1, 1, 1, 1] }
```
Comment (rengolin) [inline test/TPP/transform/transform-convolutions.mlir:343]: which conv is this going to fuse with?

Comment (chelini) [reply test/TPP/transform/transform-convolutions.mlir]: both the relus get fused into the conv.

Comment (rengolin) [reply test/TPP/transform/transform-convolutions.mlir]: relus?

Comment (chelini) [reply test/TPP/transform/transform-convolutions.mlir]: yes, relus. There are two relus and two convs. The consumer relu is fused into the conv,

Comment (rengolin) [reply test/TPP/transform/transform-convolutions.mlir]: the example doesn't exist anymore, but it had two convs and an add, and that's fusing the add:
```
%add_packed = transform.structured.map_linalg_to_tpp filter{["tpp.add"]} in %5
%6, %loop:5 = transform.structured.fuse %add_packed { tile_sizes = [1, 1, 1, 1, 1] }
```
What am I missing?

Comment (chelini) [reply test/TPP/transform/transform-convolutions.mlir]: I see. I removed the example as I was not interesting. It was fusing the convs with the add. The other example fuses the relu into the conv.

**Pull request merged**